networker/networker.go

```diff
@@ -1482,6 +1492,54 @@ func (netw *Combined) allowFileshare(publicKey string, address netip.Addr) error
 	return nil
 }
 
+func (netw *Combined) undenyDNS() error {
+	ruleName := "deny-private-dns"
+
+	ruleIndex := slices.Index(netw.rules, ruleName)
+
+	if ruleIndex == -1 {
+		return nil
+	}
+
+	if err := netw.fw.Delete([]string{ruleName}); err != nil {
+		return err
+	}
+	netw.rules = slices.Delete(netw.rules, ruleIndex, ruleIndex+1)
+
+	return nil
+}
+
+func (netw *Combined) denyDNS() error {
+	ruleName := "deny-private-dns"
+	rules := []firewall.Rule{{
+		Name:           ruleName,
+		Direction:      firewall.Outbound,
+		Protocols:      []string{"udp", "tcp"},
+		Ports:          []int{53},
```

Comment:
what should it happen if user has port `53` in allowed list?

Reply:
Allowlist FW rule will be added down after deny-DNS-private-LAN-ranges - it will work but only for non-private LAN ranges.

Reply:
Rearranged order of rules: ACCEPT rules now comes first, so, if user allowlist port 53 - that rule will be added in front of DENY rules which prevent leaks. User gets possibility to use port 53 for his custom DNS handling setup.

Reply:
Also added condition check if port 53 is whitelist then do not add drop-dns rules